### ISSUE TITLE
Add upload support

### ISSUE THREAD

**Issue by DavidePastore** [Ale46/Mega-Java#4]
Can you add the upload support?

**Comment by belluccifranco** [comment]
Would be great!

**Comment by Ale46** [comment]
With the [official sdk](https://mega.co.nz/#sdk) released by Mega this project is deprecated so adding new features is pretty useless. A java wrapper should be written instead. Anyway I set the status to this issue as "enhancement" if someone want to develop the upload function.

**Comment by DavidePastore** [comment]
@Ale46 I don't think this project is really deprecated. Mega didn't create any Java SDK, so this project could be intended to be a wrapper.

**Comment by fire855** [comment]
I tried to add it but I don't understand the encryption.. Maybe you can add it?

**Comment by cosecantt** [comment]
Waiting for upload feature soon!

**Comment by Ale46** [comment]
@cosecantt you should take a look a the official mega sdk that recently added [java bindings](https://github.com/meganz/sdk/tree/master/bindings/java) (with upload support, obviously).

**Comment by fire855** [comment]
Yes, I'm using the official mega sdk. It's working good.

**Comment by mntgoat** [comment]
@Ale46 do you know if anyone has gotten their official SDK working on Android?

**Comment by Ale46** [comment]
@mntgoat I don't have a certain answer for your question because I never used their SDK in an Android app. Anyway the official SDK have an [example app](https://github.com/meganz/sdk/tree/master/examples/android) for Android, so I assume that should work, also, for this platform.

**Comment by mntgoat** [comment]
@Ale46 thanks for the link, that should work.

**Comment by LoneDev6** [comment]
Hello @Ale46 
Any news on this feature?

**Comment by gcclinux** [comment]
am I wrong but is the SDK not only C++ based? I started t write a java backup tool and planning to use MEGA as a target, the same way I did it with the bash backup script using mega-cmd.
Has Mega drop support for Java?

**Comment by LoneDev6** [comment]
> 
> 
> am I wrong but is the SDK not only C++ based? I started t write a java backup tool and planning to use MEGA as a target, the same way I did it with the bash backup script using mega-cmd.
> Has Mega drop support for Java?

Well, there is a Java example but I didn't try it https://github.com/meganz/sdk/tree/master/examples/java

I doubt they keep example of removed API methods

**Comment by Ale46** [comment]
Hi @LoneDev6, with the bindings provided by the official mega SDK, I lost interest to keep developing this library. I agree that having a "pure" Java solution of the mega's api would be really convenient but at the same time I don't want to put some extra effort in something that, in some way, is already available and officially supported.

**Comment by LoneDev6** [comment]
> 
> 
> Hi @LoneDev6, with the bindings provided by the official mega SDK, I lost interest to keep developing this library. I agree that having a "pure" Java solution of the mega's api would be really convenient but at the same time I don't want to put some extra effort in something that, in some way, is already available and officially supported.

No worries! I understand your point.
Thanks for your hard work on the previous updates
